skada/utils.py

```diff
@@ -243,3 +243,56 @@ def source_target_split(
     return list(chain.from_iterable(
         (a[source_idx], a[~source_idx]) for a in arrays
     ))
+
+
+def source_target_merge(
```

Comment:
I think we should match split API: accept as many pairs of 'things' as necessary, alongside with a named argument 'sample_domain'. 

So I can do something like

```python 
X, y = source_target_merge(X_source, X_target, y_source, y_target, sample_domain=sample_domain)
```

Reply:
Do you mean : 

```python
X, y, sample_domain = source_target_merge(X_source, X_target, y_source, y_target)
```

insterad?

Reply:
In `source_target_split()` we don't output back the sample domain, just the splitted arrays. Makes sense to do the same here

Reply:
Right, as we have this function to pair up with the split functionality, we shouldn't encourage swapping sample_domain to a new one, even though it's technically possible (it would mean that we have to return it with every operation). Moreover, it might easily lead to 'accidental' loss of the information when having multiple sources/targets.